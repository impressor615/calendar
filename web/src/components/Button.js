import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


const Button = ({ children, color, onClick, active, ...rest }) => (
  <button
    type="button"
    onClick={onClick}
    {...rest}
    className={classnames(
      'btn',
      rest.className,
      { [color]: color },
      { active },
    )}>
    { children }
  </button>
);

Button.defaultProps = {
  color: '',
  active: false,
  onClick: () => {},
};

Button.propTypes = {
  color: PropTypes.oneOf(['', 'primary', 'secondary']),
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  active: PropTypes.bool,
}

export default Button;
