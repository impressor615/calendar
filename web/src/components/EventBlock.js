import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Button from 'components/Button';


const EventBlock = ({
  children,
  onToggle,
  onDragStart,
  fluid,
  ...rest
}) => (
  <Button
    {...rest}
    draggable
    onClick={onToggle}
    onDragStart={onDragStart}
    className={classnames(
      'event-block',
      rest.className,
      { fluid },
    )}
  >
    { children }
  </Button>
);

EventBlock.defaultProps = {
  fluid: false,
};

EventBlock.propTypes = {
  children: PropTypes.node.isRequired,
  onToggle: PropTypes.func.isRequired,
  onDragStart: PropTypes.func.isRequired,
  fluid: PropTypes.bool,
}

export default EventBlock;