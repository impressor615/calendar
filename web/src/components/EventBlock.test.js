import React from 'react';

import Button from 'components/Button';
import EventBlock from 'components/EventBlock';


describe("<EventBlock />", () => {
  const onToggle = jest.fn();
  const onDragStart = jest.fn();
  const props = {
    children: '09:00 Schedule1',
    fluid: false,
    onToggle,
    onDragStart,
  };

  it('should be rendered properly', () => {
    const wrapper = shallow(<EventBlock {...props} />);
    expect(wrapper.find('.event-block')).toHaveLength(1);
    expect(wrapper.find(Button)).toHaveLength(1);

    wrapper.setProps({ fluid: true });
    expect(wrapper.find('.event-block.fluid')).toHaveLength(1);
  });

  it('methods should be called when clicking the EventBlock', () => {
    const wrapper = shallow(<EventBlock {...props} />);
    wrapper.find(Button).simulate('click');
    wrapper.find(Button).simulate('dragstart');
    expect(onToggle).toHaveBeenCalledTimes(1);
    expect(onDragStart).toHaveBeenCalledTimes(1);
  });

  it('should be matched with the snapshot', () => {
    const wrapper = shallow(<EventBlock {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
