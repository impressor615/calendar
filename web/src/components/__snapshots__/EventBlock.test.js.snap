// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventBlock /> should be matched with the snapshot 1`] = `
<Button
  active={false}
  className="event-block"
  color=""
  draggable={true}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  onDragStart={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  09:00 Schedule1
</Button>
`;
